lib/hexpm/security_vulnerability/updater.ex

```diff
@@ -0,0 +1,141 @@
+defmodule Hexpm.SecurityVulnerability.Updater do
+  @moduledoc false
+
+  use GenServer
+
+  require Logger
+
+  alias Hexpm.Repository.Packages
+  alias Hexpm.SecurityVulnerability.Disclosures
+
+  @disclosure_download_url "https://osv-vulnerabilities.storage.googleapis.com/Hex/all.zip"
+
+  @enforce_keys [:update_interval]
+  defstruct [:update_interval, etag: nil]
+
+  @start_opts [
+    :timeout,
+    :debug,
+    :spawn_opt,
+    name: __MODULE__,
+    hibernate_after: to_timeout(second: 1)
+  ]
+  @start_keys [:start, :timeout, :debug, :spawn_opt, :hibernate_after]
+  @init_opts [update_interval: to_timeout(minute: 30)]
```

Comment:
This pulls down the zip file every 30 minutes? IMO that is overkill.
The last vulnerability identified was yesterday, but the one before that was 6 months ago.
I'd like to stay on top of things, but that hit rate seems pretty low.

<img width="473" alt="Screenshot 2024-10-25 at 6 43 59 AM" src="https://github.com/user-attachments/assets/ef6bea89-99d6-4710-8ab4-aa15b5811449">

Reply:
I chose the interval so that a vulnerability is recognized and propagated on quickly. Since it uses the ETag to avoid reading the same file multiple times, this should be relatively inexpensive.